### ISSUE TITLE
NVDA fails to report some list items

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#15023]
### Steps to reproduce:
1. Open the page: https://www.solidstatelogic.com/where-to-buy
2. Press letter B navigate to "Select a product"
3. press Space, then press Downarrow;
### Actual behavior:
NVDA Silence.
### Expected behavior:
NVDA can report list items, just like Narrator
### NVDA logs, crash dumps and other attachments:
[nvda_log.txt](https://github.com/nvaccess/nvda/files/11783652/nvda_log.txt)

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
alpha-28447
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.3086
#### Name and version of other software in use when reproducing the issue:
Chrome 114.0.5735.134
#### Other information about your system:
No
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA2023.1 can also reproduce
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by thgcode** [comment]
I think in this case, NVDA switches to focus mode to pass the keyboard keys to the combobox, but Narrator keeps in browse mode. The question is why it doesn't let go back to browse mode. I'll investigate this further to see if I discover anything more.

**Comment by thgcode** [comment]
I think it could be some code on the virtual buffer system. On the gecko_ia2.cpp file, there is some code that disables rendering of child elements if they are inside a combobox. This might be the cause why browse mode stops working when we press spacebar on the button. Focus moves inside the list and since the browse mode virtual buffer doesn't render it, browse mode breaks.

**Comment by cary-rowen** [comment]
Thanks to @thgcode for the investigation, @jcsteh Can you share your thoughts on this?

**Comment by jcsteh** [comment]
This is due to authoring error. Per the ARIA spec, a list box should manage focus; i.e. the arrow keys should move focus between list items. This list box doesn't handle the arrow keys at all.

It works in Narrator because Narrator remains in scan mode and chooses to render list box items in scan mode. However, that causes its own problems. For example, if there is a list box containing 1000 items, you'll have to cursor through all of them in scan mode (or jump via some other means) to get past the list box, which is why NVDA doesn't render list box items in browse mode.

You can get around this authoring error in NVDA by navigating objects using the review cursor.

**Comment by seanbudd** [comment]
Closing due to authoring error.
As a starting point please submit feedback to the authors to let them know about the problem.

Kind Regards,
NV Access Software Developers
